openspoor/spoortakmodel/spoortak_models_data.py

```diff
@@ -0,0 +1,103 @@
+import os
+from glob import glob
+
+import numpy as np
+from typing import Optional
+import logging
+import pandas as pd
+
+from ..spoortakmodel.singleton import Singleton
+
+log = logging.getLogger(__name__)
+
+
+class SpoortakModelsData(Singleton):
+    """ Helper class that loads spoortak model data (only once)
+
+    Improvement backlog:
+    - Support V18 (spoortak 2.0)
+    """
+
+    models: dict = None
+    model_changes: dict = None
+
+    @staticmethod
+    def _convert_dutch_number(value: str) -> float:
+        """ '55.132,56' => 55132.56"""
+        if not value:
+            return np.nan
+        return float(value.replace('.', '').replace(',', '.'))
+
+    @staticmethod
+    def _km_to_meters(km: float) -> Optional[int]:
+        if km is None or np.isnan(km):
+            return None
+        return int(km * 1000)
+
+    @staticmethod
+    def _replace_km_columns(df: pd.DataFrame) -> None:
+        """ implace correction of the km columns and renaming them to avoid confusion """
+        df['kilometrering_start'] = df[['KM_BEGIN', 'KM_EIND']].values.min(1)
+        df['kilometrering_end'] = df[['KM_BEGIN', 'KM_EIND']].values.max(1)
+
+        df_changed = df[df['kilometrering_start'] != df['KM_BEGIN']]
+        df_changed_geo = df_changed[df_changed['GEOCODE_BEGIN'] != df_changed['GEOCODE_EIND']]
+
+        for spoortak, row in df_changed_geo.iterrows():
+            log.warning(
+                f'Swapped begin & eind km for {spoortak} met verschillende geocode begin & eind mogenlijk was dit een ander kilometerlint')
+
+        df.drop(columns=['KM_BEGIN', 'KM_EIND'], inplace=True)
+
+    @staticmethod
+    def _get_model_numbers(data_path: str) -> [int]:
+        """ Return the available model numbers """
+        dirs = glob(os.path.join(data_path, 'Versie_*'))
+
+        # return [int(os.path.basename(directory).removeprefix('Version_')) for directory in dirs] # python 3.9 feature
+        return [int(os.path.basename(directory).split('_')[1]) for directory in dirs]
```

Comment:
The pathlib library may have a more readable alternative: probably something like
from pathlib import Path
Path(directory).name.split('_')[1]

Reply:
does not reduce the 'complexity' it is still a similar function call.
But I'll just import basename so that we can drop the os.path (will do the same for join in this case) to make it a little bit more readable. Als added a bit more info in de docstring